op-batcher/batcher/channel_manager.go

```diff
@@ -190,23 +193,44 @@ func (s *channelManager) TxData(l1Head eth.BlockID) (txData, error) {
 	return s.nextTxData(s.currentChannel)
 }
 
+func (s *channelManager) getChannelStartBlock(l1SafeHead eth.BlockID) eth.BlockID {
+	if s.l1OriginLastClosedChannel != nil {
+		return *s.l1OriginLastClosedChannel
+	}
+
+	// If there is no last closed channel, we should start from the safe head
+	return l1SafeHead
+}
+
 // ensureChannelWithSpace ensures currentChannel is populated with a channel that has
 // space for more data (i.e. channel.IsFull returns false). If currentChannel is nil
 // or full, a new channel is created.
-func (s *channelManager) ensureChannelWithSpace(l1Head eth.BlockID) error {
+func (s *channelManager) ensureChannelWithSpace(l1Head eth.BlockID, l1SafeHead eth.BlockID) error {
 	if s.currentChannel != nil && !s.currentChannel.IsFull() {
 		return nil
 	}
 
 	pc, err := newChannel(s.log, s.metr, s.cfg, s.rollupCfg)
+
 	if err != nil {
 		return fmt.Errorf("creating new channel: %w", err)
 	}
+
+	startBlock := s.getChannelStartBlock(l1SafeHead)
+	if startBlock.Number > 0 {
+		// Sets the timeout of the channel to be the latest safe head + max channel timeout
+		pc.RegisterL1Block(startBlock.Number)
+	}
```

Comment:
In relation to my other comment about simplifying this, we'd then here just do
```suggestion
	startBlock := s.l1OriginLastClosedChannel
	if startBlock.Number > 0 {
		// Sets the timeout of the channel to be the latest safe head + max channel timeout
		pc.RegisterL1Block(startBlock.Number)
	}
```
and could delete method `getChannelStartBlock`.

We'd be guaranteed to have a value for `l1OriginLastClosedChannel` because we create it once at startup (or after reorgs/resets).